### ISSUE TITLE
Use a GitHub Action to automate and centralise management of our GitHub Teams

### ISSUE THREAD

**Issue by sgibson91** [2i2c-org/team-compass#768]
### Context

I was recently introduced to this GitHub Action, used by the Scientific Python community, that can automatically sync GitHub Teams: https://github.com/scientific-python/sync-teams-action I think it would be beneficial to us to adopt it for a few reasons:

- Centralised location/single source of truth of what teams exist, who is a part of them, and what permissions they have
- When a person was added to/removed from a team and by whom will be captured in git history

NOTE: The org name is hard-coded in that repo, so I may see if they'll accept a PR to make it configurable

QUESTION: Would we prefer our team.yaml file defined in a public repo (team-compass) or a private repo (meta)?

**Comment by damianavila** [comment]
> NOTE: The org name is hard-coded in that repo, so I may see if they'll accept a PR to make it configurable

Makes sense to me.

> QUESTION: Would we prefer our team.yaml file defined in a public repo (team-compass) or a private repo (meta)?

Mmm... that is a good question.
My reactive answer would be to protect it under meta.
Wondering what others think about it...

**Comment by sgibson91** [comment]
> My reactive answer would be to protect it under meta.

Yes, it can be moved to team-compass later if agreed upon. But if I start it in team-compass and then it is agreed it should be in meta, cleaning up the data to maintain privacy is much more difficult.
